modules/apps/ccv/types/expected_keepers.go

```diff
@@ -0,0 +1,17 @@
+package types
+
+// StakingKeeper defines the contract expected by parent-chain ccv module.
+// StakingKeeper is responsible for keeping track of latest validator set of all baby chains
+type StakingKeeper interface {
+	GetValidatorSetChanges(chainID string) []abci.ValidatorUpdate
+	// This method is not required by CCV module explicitly but necessary for init protocol
+	GetInitialValidatorSet(chainID string) []sdk.Tx
+}
+
+// CNSKeeper defines the contract expected by parent-chain ccv module
+// CNSKeeper is responsible for verifying that chain creator is authorized to create a chain with given chain-id
```

Comment:
It took me a couple minutes to realize you were talking about the chain name service / registry
```suggestion
// The Chain Name Service (CNS) is responsible for verifying that the chain creator is authorized to register a chain with the given identifier
```

Reply:
Will change to Chain Name Service. However CCV is asking CNS if the creator can call `SubmitNewChain` with this chainID. For CNS to authorize successfully, the creator must have already registered the chainID